dev/plots_api_strawman.py

```diff
@@ -0,0 +1,242 @@
+    location=my_location,
+    pressure=0.615 * u.bar,
+    relative_humidity=0.11,
+    temperature=0 * u.deg_C,
+    timezone=timezone('US/Hawaii'),
+    description="Subaru Telescope on Mauna Kea, Hawaii"
+    )
+
+# Or call up one of of the sites in Astroplan's database.
+
+from astroplan import sites
+
+obs = sites.Keck1
+
+# Then, define your target.
+
+from astroplan import FixedTarget
+from astroplan.coordinates import SkyCoord
+
+my_target = FixedTarget(
+    name='Polaris',
+    coord=SkyCoord('02h31m49.09s', '+89d15m50.8s', frame='icrs')
+    )
+
+# You can also define an array with multiple targets.
+targets = [
+    FixedTarget(name='Polaris',
+                coord=SkyCoord('02h31m49.09s', '+89d15m50.8s', frame='icrs')),
+    FixedTarget(name='Sirius',
+                coord=SkyCoord('06h45m08.9173s', '-16d42m58.017s', frame='icrs')),
+    FixedTarget(name='Other Target',
+                coord=SkyCoord('01h30m0s', '+10d15m20s', frame='icrs')),
+    ]
+
+# ...once you've constructed your Observer and Target objects, you can plot any
+#   any number of quantities using the plotting functions and Time objects.
+
+
+# ===============
+# Airmass vs Time
+# ===============
+
+from astroplan import plot_airmass
+from astropy.time import Time
+import numpy as np
+
+# plot_airmass takes an individual Target, an individual Observer,
+#   and a Time object as arguments.
+
+# Construct the Time object with as many individual times as you want data
+#   points in the resulting airmass plot.
+
+# One way to do this is with a TimeDelta object and numpy.linspace().
+
+start_time = Time('2015-01-15T21:00:00')
+end_time = Time('2015-01-16T04:00:00')
+delta_t_airmass = end_time - start_time
+times_airmass = start_time + delta_t_airmass*np.linspace(0.0, 1.0, 100)
+
+plot_airmass(my_target, observer=obs, time=times_airmass)
+
+# If you want to populate a Time object starting with just one time,
+# you can use astropy.units.
+
+times_airmass = start_time + np.linspace(-8, 8, 100)*u.hour
+
+# If you want to plot airmass for multiple targets on the same plot,
+#   use the new_plot=False option for each target after the first
+#   (the default is True).
+
+plot_airmass(targets[0], observer=obs, time=times_airmass)
+plot_airmass(targets[1], observer=obs, time=times_airmass, new_plot=False)
+plot_airmass(targets[2], observer=obs, time=times_airmass, new_plot=False)
+
+# If you want to plot airmass for multiple targets on different plots,
+#   simply issue the command as normal.
+
+for i in range(0, len(targets)):
+    plot_airmass(targets[i], observer=obs, time=times_airmass)
+
+# You can also set a dark background with the dark_plot=True option.
+
+plot_airmass(my_target, observer=obs, time=times_airmass, dark_plot=True)
+
+# For custom line styles/colors, use the style_kwargs option in the following
+#   manner.
+
+plot_airmass(
+    my_target,
+    observer=obs,
+    time=times_airmass,
+    style_kwarg={'linestyle': '--', 'color': 'r'},
```

Comment:
@cdeil - re: the error question: it's ok with ``**kwargs`` because they get sent directly to matplotlib which would give errors for invalid keywords.  You're right it's a little less clear because it would come from `matplotlib` instead of the `astorplan.plot_*` function, though.

As for multiple plotting functions, @jberlanga, @cdeil means that if there are *two* calls to matplotlib inside a `astroplan.plot_*` function, using ``**kwarg`` you can only unambiguously send the keywords off to one of them.  For example if we had a `plot_airmass_and_moon` plot, you'd need to make *two* calls to ``scatter`` or ``plot``, and if the style stuff is in a dictionary, you can just make two dictionaries: ``airmass_style_kwargs`` and ``moon_style_kwargs``.  But if the keywords are forwarded directly using ``**kwargs``, it can only be done for one of them. I agree that *might* be an issue, but it might be that we agree now to not do more than one plot per function call?  (I.e., require the user to do ``plot_airmass`` and ``plot_moon`` as separate functions).